src/journal-remote/journal-gatewayd.c

```diff
@@ -307,42 +345,99 @@ static int request_parse_range(
         if (!colon)
                 m->cursor = strdup(range);
         else {
-                const char *p;
+                r = request_parse_range_skip_and_n_entries(m, colon);
+                if (r < 0)
+                        return r;
 
-                colon2 = strchr(colon + 1, ':');
-                if (colon2) {
-                        _cleanup_free_ char *t;
+                m->cursor = strndup(range, colon - range);
+        }
 
-                        t = strndup(colon + 1, colon2 - colon - 1);
-                        if (!t)
-                                return -ENOMEM;
+        if (!m->cursor)
+                return -ENOMEM;
 
-                        r = safe_atoi64(t, &m->n_skip);
-                        if (r < 0)
-                                return r;
-                }
+        m->cursor[strcspn(m->cursor, WHITESPACE)] = 0;
+        if (isempty(m->cursor))
+                m->cursor = mfree(m->cursor);
 
-                p = (colon2 ? colon2 : colon) + 1;
-                if (*p) {
-                        r = safe_atou64(p, &m->n_entries);
-                        if (r < 0)
-                                return r;
+        return 0;
+}
 
-                        if (m->n_entries <= 0)
-                                return -EINVAL;
 
-                        m->n_entries_set = true;
-                }
+static int request_parse_range_time(
+                RequestMeta *m,
+                const char *range) {
 
-                m->cursor = strndup(range, colon - range);
+        char *colon, *until;
+        int r;
+
+        range += 5;
+        range += strspn(range, WHITESPACE);
+
+        colon = strchr(range, ':');
+        if (!colon)
+                return -EINVAL;
+
+        if (colon - range > 0) {
+                _cleanup_free_ char *t;
+                t = strndup(range, colon - range);
+                if (!t)
+                        return -ENOMEM;
+
+                r = safe_atou64(t, &m->since);
```

Comment:
Maybe, `parse_usec()` is better?

Reply:
I've not found parse_usec function, only a target for jumps in [basic/time-util.c](https://github.com/systemd/systemd/blob/main/src/basic/time-util.c#L822). I'm sorry, I don't know how to use it (I'm really beginning with C by this code). Could you precise your hint ?

Reply:
Ah, sorry, `parse_sec()`.